spec/active_hash/base_spec.rb

```diff
@@ -311,6 +323,19 @@ class Region < ActiveHash::Base
       record.id.should == 1
       record.name.should == 'US'
     end
+
+    it "finds the record with the specified id as a string" do
+      record = Country.find_by(:id => '1')
+      record.name.should == 'US'
+    end
+
+    it "returns the record that matches options" do
+      expect(Country.find_by(:name => "US").id).to eq(1)
```

Comment:
looks like `find_by` and `where` and others do not work with string keys. Find by `"id"` does seem to work as a string.

Seems like a separate feature request